pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -387,8 +388,18 @@ func DestroyCluster(l *logger.Logger, c *Cluster) error {
 		if _, ok := promhelperclient.SupportedPromProjects[node.Project]; ok &&
 			node.Provider == gce.ProviderName {
 			if err := promhelperclient.NewPromClient().DeleteClusterConfig(context.Background(),
-				c.Name, false, l); err != nil {
-				l.Errorf("Failed to delete the cluster config: %v", err)
+				c.Name, false, false /* insecure */, l); err != nil {
+				// TODO(bhaskar): Obtain secure cluster information.
+				// Cluster does not have the information on secure or not. So, we retry as insecure
+				// if delete fails with cluster as secure
+				if strings.Contains(err.Error(), "request failed with status 404") {
```

Comment:
In hindsight, we should have exposed `response.StatusCode`. String matching on the response payload makes refactoring more error-prone.